src/config.rs

```diff
@@ -233,7 +233,9 @@ impl Config {
         };
 
         let magic = match (config.network, config.signet_magic) {
-            (Network::Signet, Some(magic)) => magic,
+            (Network::Signet, Some(magic)) => u32::from_str_radix(&magic, 16)
+                .expect("Invalid signet hex magic")
```

Comment:
It'd be much better to have a user-friendly error message:

```rust
.unwrap_or_else(|error| {
    eprintln!("Error: the signet magic '{}' is not a valid hex string: {}", magic, error);
    std::process::exit(1);
})
```

Reply:
Hi, it will probably clearer this way in case user messes up, thanks